onnxruntime/core/providers/xnnpack/nn/average_pool.cc

```diff
@@ -241,9 +245,12 @@ Status AveragePool::Compute(OpKernelContext* context) const {
 
   std::unique_ptr<void, decltype(deallocator)> workspace(nullptr, deallocator);
 
-  auto reshape_fn = (avgpool_type_ == OpComputeType::op_compute_type_fp32)
-                        ? xnn_reshape_average_pooling2d_nhwc_f32
-                        : xnn_reshape_average_pooling2d_nhwc_qu8;
+  auto reshape_fn = xnn_reshape_average_pooling2d_nhwc_f32;
+  if (avgpool_type_ == OpComputeType::op_compute_type_fp16) {
+    reshape_fn = xnn_reshape_average_pooling2d_nhwc_f16;
+  } else {
```

Comment:
else if (....= qu8?

Otherwise `xnn_reshape_average_pooling2d_nhwc_f32` won't be selected anymore.

Reply:
Good catch. Do we not have a unit test that should have failed due to this?

Reply:
The issue isn't that there isn't a unit test.
The issue is that the test passed but not really executed.
If the capabilities is empty, it still returns OK.
https://github.com/microsoft/onnxruntime/blob/72cc72cc21249bfa91c6564edc341e3aa881e542/onnxruntime/core/framework/graph_partitioner.cc#L177-L179.
As a result, if one kernel's IsONNXNodeSupported return false, the kernel will not be executed, but the tests still return passed.
I had added a throw in compute() to check if one kernel is really executed. The averagePool was added last, I forgot to check it and there's an error in IsONNXNodeSupported. So, The test passed but not really executed. In fact, the test can catch this mistake.
We may need to redesign some logic in unit test workflow in the next step. 
In the past, even the F32 XNNPACK matmul tests are never really executed. So I added 2 new tests in this PR.